src/main/java/core/basesyntax/dao/animal/AnimalDaoImpl.java

```diff
@@ -12,11 +16,35 @@ public AnimalDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Animal save(Animal animal) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(animal);
+            transaction.commit();
+            return animal;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add an animal " + animal
+                    + " in a DB!", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Animal> findByNameFirstLetter(Character character) {
-        return null;
+        try (Session session = sessionFactory.openSession()) {
+            Query<Animal> animalQuery = session.createQuery("FROM Animal WHERE name LIKE '"
+                    + character.toString().toLowerCase() + "%'"
+                    + "OR name LIKE '" + character.toString().toUpperCase()
+                    + "%'", Animal.class);
```

Comment:
```suggestion
            Query<Animal> getAnimalByFirstLetter = session.createQuery("FROM Animal "
                    + "WHERE UPPER(name) LIKE CONCAT(:character, '%')", Animal.class);
            getAnimalByFirstLetter.setParameter("character", character);
```

I think it's better not to use concatenation